modules/database/database-sql/src/test/java/org/eclipse/dirigible/database/sql/dialects/hana/HanaCreateTableBuilderTest.java

```diff
@@ -0,0 +1,217 @@
+        .columnInteger("ID2", true, false, false)
+        .columnVarchar("FIRST_NAME", 20, false, true, true)
+        .columnVarchar("LAST_NAME", 30, false, true, false)
+        .primaryKey(new String[]{"ID", "ID2"})
+        .build();
+
+    assertNotNull(sql);
+    assertEquals("CREATE COLUMN TABLE CUSTOMERS ( ID INTEGER NOT NULL , ID2 INTEGER NOT NULL , FIRST_NAME VARCHAR (20) UNIQUE , LAST_NAME VARCHAR (30) , PRIMARY KEY(ID , ID2) )", sql);
+  }
+
+  @Test
+  public void createTableWithSetPKOnConstraintAndColumnLevel() {
+    String sql = SqlFactory.getNative(new HanaSqlDialect())
+        .create()
+        .columnTable("CUSTOMERS")
+        .columnInteger("ID", true, false, false)
+        .columnVarchar("FIRST_NAME", 20, false, true, true)
+        .primaryKey(new String[]{"ID"})
+        .build();
+
+    assertNotNull(sql);
+    assertEquals("CREATE COLUMN TABLE CUSTOMERS ( ID INTEGER NOT NULL PRIMARY KEY , FIRST_NAME VARCHAR (20) UNIQUE )", sql);
+  }
+
+  @Test
+  public void parseTableWithoutPK() {
+    String sql = SqlFactory.getNative(new HanaSqlDialect())
+        .create()
+        .columnTable("CUSTOMERS")
+        .columnInteger("ID", false, false, false)
+        .columnVarchar("FIRST_NAME", 20, false, true, true)
+        .primaryKey(new String[]{})
+        .build();
+
+    assertNotNull(sql);
+    assertEquals("CREATE COLUMN TABLE CUSTOMERS ( ID INTEGER NOT NULL , FIRST_NAME VARCHAR (20) UNIQUE )", sql);
+  }
+
+  /**
+   * Creates the row table with indexes.
+   */
+  @Test
+  public void createRowTableWithIndexes() {
+    String sql = SqlFactory.getNative(new HanaSqlDialect())
+        .create()
+        .rowTable("CUSTOMERS")
+        .column("ID", DataType.INTEGER, true, false, false)
+        .column("FIRST_NAME", DataType.VARCHAR, false, false, true, "(20)")
+        .column("LAST_NAME", DataType.VARCHAR, false, true, false, "(30)")
+        .index("I1", false, "DESC", "BTREE", new HashSet<>(List.of("LAST_NAME")))
+        .unique("I2", new String[] {"ID"}, "CPBTREE", "ASC")
+        .build();
+
+    assertNotNull("Unexpected result from builder", sql);
+    assertTrue("Expected create table statement was not found", sql.contains("CREATE ROW TABLE CUSTOMERS ( ID INTEGER NOT NULL PRIMARY KEY , FIRST_NAME VARCHAR (20) NOT NULL UNIQUE , LAST_NAME VARCHAR (30) );"));
+    assertTrue("Expected unique index statement was not found", sql.contains("CREATE UNIQUE CPBTREE INDEX I2 ON CUSTOMERS ( ID ) ASC"));
+    assertTrue("Expected index statement was not found", sql.contains("CREATE BTREE INDEX I1 ON CUSTOMERS ( LAST_NAME ) DESC"));
+    int expectedStatementLength = "CREATE ROW TABLE CUSTOMERS ( ID INTEGER NOT NULL PRIMARY KEY , FIRST_NAME VARCHAR (20) NOT NULL UNIQUE , LAST_NAME VARCHAR (30) ); CREATE UNIQUE CPBTREE INDEX I2 ON CUSTOMERS ( ID ) ASC; CREATE BTREE INDEX I1 ON CUSTOMERS ( LAST_NAME ) DESC".length();
+    assertEquals("Unexpected length of statement", expectedStatementLength, sql.length());
```

Comment:
Can this be simplified as follows
```java
String expectedStatement = "CREATE ROW TABLE CUSTOMERS ( ID INTEGER NOT NULL PRIMARY KEY , FIRST_NAME VARCHAR (20) NOT NULL UNIQUE , LAST_NAME VARCHAR (30) ); CREATE UNIQUE CPBTREE INDEX I2 ON CUSTOMERS ( ID ) ASC; CREATE BTREE INDEX I1 ON CUSTOMERS ( LAST_NAME ) DESC";
assertEquals("Unexpected statement", expectedStatement, sql);
```

Reply:
No, because the order of the indices is not guaranteed and the test will not be stable.